portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,10 +34,55 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private String name;
+  ArrayList<String> comments = new ArrayList<String>();
+  List<Task> tasks = new ArrayList<>();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = "Bless";
+    String json = convertToJson();
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String title = (String) entity.getProperty("title");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Task task = new Task(id, title, timestamp);
+      tasks.add(task);
+    }
+  }
+  private String convertToJson() {
+    Gson gson = new Gson();
+    String json = gson.toJson(tasks);
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
```

Comment:
Would recommend having this as a private static final String class constant and used through, to prevent copy-paste errors (ie. Julien's comment about title) and ensuring that the same key is used everywhere.  The field name should be all caps with underscore (TEXT_INPUT)

Reply:
Can you please clarify a bit more?

Reply:
Similar to how you have:

```ArrayList<String> comments = new ArrayList<String>();```

You can also declare:

```private static final String TEXT_INPUT = "text-input";```

and use that variable throughout.

Reply:
Ohhhhh okay got it!